src/Token/BedrockTokenProvider.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Aws\Token;
+
+use Aws\Configuration\ConfigurationResolver;
+use Aws\Exception\TokenException;
+use GuzzleHttp\Promise;
+
+/**
+ * Token provider for Bedrock that sources bearer tokens from environment variables.
+ */
+class BedrockTokenProvider extends TokenProvider
+{
+    public const TOKEN_ENV_KEY = 'bearer_token_bedrock';
+    public const BEARER_AUTH = 'smithy.api#httpBearerAuth';
+
+    /**
+     * Create a default Bedrock token provider that checks for a bearer token
+     * in the AWS_BEARER_TOKEN_BEDROCK environment variable.
+     *
+     * This provider is automatically wrapped in a memoize function that caches
+     * previously provided tokens.
+     *
+     * @param array $config Optional array of token provider options.
+     *
+     * @return callable
+     */
+    public static function defaultProvider(array $config = []): callable
```

Comment:
On this implementation, what would be the usage of this function?. I could not see where it is used.